models/documentlistmodel.cpp

```diff
@@ -125,6 +143,35 @@ void DocumentListModel::addItem(QString name, QString path, QString type, int si
     endInsertRows();
 }
 
+void DocumentListModel::removeItemsDirty()
+{
+    int start;
+    int index = 0, j;
+
+    start = -1;
+    for(; index < d->entries.count(); ++index) {
+        if(d->entries.at(index).dirty) {
+            if(start < 0)
+                start = index;
+        } else {
+            if(start>=0) {
+                beginRemoveRows(QModelIndex(), start, index - 1);
+                for (j = start; j < index; j++)
+                    d->entries.removeAt(j);
+                endRemoveRows();
+                start = -1;
+            }
+        }
+    }
+    if(start>=0) {
+        beginRemoveRows(QModelIndex(), start, index - 1);
+        for (j = start; j < index; j++)
+            d->entries.removeAt(j);
+        endRemoveRows();
+    }
```

Comment:
The way above it will send signals only when a entry will be removed, please note I missed a if there, will update.